File: sdks/csharp/src/DarkStrata.CredentialCheck/Exceptions/AuthenticationException.cs
namespace DarkStrata.CredentialCheck;

/// <summary>
/// Exception thrown when API key authentication fails.
/// </summary>
public sealed class AuthenticationException : DarkStrataException
{
    /// <summary>
    /// Creates a new AuthenticationException.
    /// </summary>
    /// <param name="message">The error message.</param>
    public AuthenticationException(string message = "Invalid or missing API key")
        : base(message, ErrorCode.AuthenticationError, statusCode: 401, isRetryable: false)
    {
    }
}

File: sdks/java/src/main/java/io/darkstrata/credentialcheck/exception/AuthenticationException.java
package io.darkstrata.credentialcheck.exception;

/**
 * Exception thrown when API authentication fails (401).
 */
public class AuthenticationException extends DarkStrataException {

    public AuthenticationException() {
        this("Invalid or missing API key");
    }

    public AuthenticationException(String message) {
        super(message, ErrorCode.AUTHENTICATION_ERROR, 401, false);
    }
}
